libs/framework/include/celix_bundle_context.h

```diff
@@ -771,6 +771,12 @@ typedef struct celix_service_use_options {
      * and the bundle owning the service will also be provided to the callback.
      */
     void (*useWithOwner)(void *handle, void *svc, const celix_properties_t *props, const celix_bundle_t *svcOwner) OPTS_INIT;
+    /**
+     * @brief Call the provided callbacks from the caller thread directly if set, otherwise the callbacks will be called from the Celix event loop (most likely indirectly).
+     * Note that using blocking service in the Celix event loop is generally a bad idea, which should be avoided if possible.
+     */
+#define CELIX_SERVICE_USE_DIRECT          (1)
+    int flags;
```

Comment:
For C++ support the `OPTS_INIT` should be added so that all the options fields have a default initialisation.

Reply:
Ouch! It's fixed.